src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt

```diff
@@ -76,21 +67,16 @@ class CampaignServiceImpl(
       }
     })
     if (grossPrice != null) {
-      val insuranceCost = calculateInsuranceCost(member, grossPrice, bundleDiscount)
+      val insuranceCost = PremiumCalculations.calculateInsuranceCostWithAllRedeemedCampaigns(member, grossPrice, bundleDiscount)
 
       return RedeemedCodeResult(insuranceCost, campaigns)
     }
 
     return RedeemedCodeResult(null, campaigns)
   }
 
-  override fun getInsuranceCostOnPrice(memberId: String, grossPrice: Money, bundleDiscount: BundleDiscount?): InsuranceCost {
-    val member =
-      campaignOwnerRepository.findByMemberIdLoadRedeemedCampaignsWithOwnersAndReferralCampaignRedemptions(memberId)
-        ?: throw NullPointerException("Could not find campaignOwner for member $memberId")
-
-    return calculateInsuranceCost(member, grossPrice, bundleDiscount)
-  }
+  override fun getInsuranceCostOnPrice(memberId: String, grossPrice: Money, bundleDiscount: BundleDiscount?) =
+    calculateInsuranceCostWithAllRedeemedCampaigns(getMember(memberId), grossPrice, bundleDiscount)
```

Comment:
```suggestion
    PremiumCalculations.calculateInsuranceCostWithAllRedeemedCampaigns(getMember(memberId), grossPrice, bundleDiscount)
```
Maybe we should add this so that it is clear that this is a static function? Since static functions so far is quite uncommon in our code.